Doc/whatsnew/3.14.rst

```diff
@@ -311,6 +312,24 @@ For further information on how to build Python, see
 (Contributed by Ken Jin in :gh:`128563`, with ideas on how to implement this
 in CPython by Mark Shannon, Garrett Gu, Haoran Xu, and Josh Haberman.)
 
+.. _whatsnew314-free-threaded-cpython:
+
+Free-threaded CPython
+---------------------
+
+Free-threaded CPython now supports :pep:`659` which is known
+as the specializing adaptive interpreter, the execution mechanisms is almost same
+as the default build, but at free-threaded CPython build, each thread maintains
+a thread-local copy of specialized bytecode.
+
+With this change, the free-threaded CPython build achieve better performance
+and 10-15% slower than the default build.
+
+This work was done by many contributors, Matt Page, Neil Schemenauer,
+Sam Gross, Thomas Wouters, Donghee Na, Kirill Podoprigora and Ken Jin.
```

Comment:
I also added the @Fidget-Spinner, because I remember his early design involvation while he worked at Quansight.

Reply:
I think the main thing I did that is coming in 3.14 is the stackrefs.